webapp/tests/karma/unit/services/db-sync.js

```diff
@@ -10,17 +10,23 @@ describe('DBSync service', () => {
       isAdmin,
       userCtx,
       sync,
-      Auth;
+      Auth,
+      circularOn;
 
   beforeEach(() => {
+    circularOn = sinon.stub();
+    circularOn.returns({ on: circularOn });
```

Comment:
Hah. Nice. Maybe call it `recursiveOn`? This might also be useful in KarmaUtils or something - I'm sure I've wanted it before...